tableau_blueprints/download_view.py

```diff
@@ -0,0 +1,133 @@
+        server = TSC.Server(content_url, use_server_version=True)
+        server.auth.sign_in(tableau_auth)
+    except Exception as e:
+        print(f'Failed to connect to Tableau.')
+        print(e)
+        sys.exit(EXIT_CODE_INVALID_CREDENTIALS)
+    return server, tableau_auth
+
+
+def validate_get_view(server, tableau_auth, view_name):
+    """Returns the details of a specific view.
+
+    :param server:
+    :param tableau_auth:
+    :param view_name:
+    :return:
+    """
+    try:
+        view_id = None
+        with server.auth.sign_in(tableau_auth):
+            all_views, pagination_item = server.views.get()
+            for views in all_views:
+                if views.name == view_name:
+                    # view_obj = server.views.get_by_id(views.id)
+                    view_id = views.id
+    except Exception as e:
+        print(f'View item may not be valid.')
+        print(e)
+        sys.exit(EXIT_CODE_INVALID_RESOURCE)
+    return view_id
+
+
+def download_view_item(server, tableau_auth, filename, filetype, view_id):
+    """to download a view from Tableau Server
+
+    :param server:
+    :param tableau_auth:
+    :param filename: name of the view to be downloaded as.
+    :param filetype: 'image', 'thumbnail', 'pdf', 'csv'
+    :return:
+    """
+    try:
+        with server.auth.sign_in(tableau_auth):
+            views = server.views.get_by_id(view_id)
+            if filetype == "image":
+                server.views.populate_image(views)
+                with open('./' + filename, 'wb') as f:
+                    f.write(views.image)
+
+            if filetype == "pdf":  # Populate and save the CSV data in a file
+                server.views.populate_pdf(views, req_options=None)
+                with open('./' + filename, 'wb') as f:
+                    f.write(views.pdf)
+
+            if filetype == "csv":
+                server.views.populate_pdf(views, req_options=None)
+                with open('./' + filename, 'wb') as f:
+                    # Perform byte join on the CSV data
+                    f.write(b''.join(views.csv))
+
+            if filetype == "thumbnail":
+                server.views.populate_pdf(views, req_options=None)
+                with open('./' + filename, 'wb') as f:
+                    # Perform byte join on the CSV data
+                    f.write(b''.join(views.csv))
+            print("View saved to current directory successfully")
+    except OSError as e:
+        print(f'Could not write file:.')
+        print(e)
+        sys.exit(EXIT_CODE_INVALID_RESOURCE)
+
+    return True
+
+
+def main():
+    args = get_args()
+    username = args.username
+    password = args.password
+    site_id = args.site_id
+    server_url = args.server_url
+    view_name = args.view_name
+    filetype = args.file_type
+    filename = args.file_name
+    fileoptions = args.file_options  # TODO
+    server, tableau_auth = authenticate_tableau(username, password, site_id, server_url)
```

Comment:
I like the `authenticate_tableau` function, and it looks like it's being used for making sure we can authenticate and exit with a specific error code if not. But, then in the other functions you are passing in `server`, `tableau_auth` and then calling `server.auth.sign_in(tableau_auth)` multiple times in a `with` clause. 
Is that a requirement of the way the sdk works - I'm guessing not? 

Can you instead do the authentication once? You could call `authenticate_tableau` and have that function return the context to use in `with`, or call `with server.auth.sign_in(tableau_auth)` once in main, and pass just `server.views` to the other functions that need it. 
This all reduces the number of parameters needed in each function, and forces authentication to a single point. 

**This also applies to the `refresh_datasource.py` file.**